src/sentry/static/sentry/app/components/charts/baseChart.tsx

```diff
@@ -0,0 +1,455 @@
+     * triggered by toolbar
+     */
+    startValue?: number;
+    /**
+     * data value of zoom finish position; only exists in zoom event of
+     * triggered by toolbar
+     */
+    endValue?: number;
+  }>;
+  /**
+   * One example of when this is called is restoring chart from zoom levels
+   */
+  onRestore?: EChartEventHandler<{type: 'restore'}>;
+  onFinished: EChartEventHandler<{}>;
+  /**
+   * Forwarded Ref
+   */
+  forwardedRef: React.Ref<ReactEchartsCore>;
+  /**
+   * Custom chart props that are implemented by us (and not a feature of eCharts)
+   *
+   * Display previous period as a LineSeries
+   */
+  previousPeriod: Series[];
+  /**
+   * If data is grouped by date; then apply default date formatting to x-axis
+   * and tooltips.
+   */
+  isGroupedByDate?: boolean;
+  /**
+   * Format timestamp with date AND time
+   */
+  showTimeInTooltip?: boolean;
+  /**
+   * Use short date formatting for xAxis
+   */
+  useShortDate?: boolean;
+  /**
+   * optional, used to determine how xAxis is formatted if `isGroupedByDate == true`
+   */
+  start?: Date;
+  /**
+   * optional, used to determine how xAxis is formatted if `isGroupedByDate == true`
+   */
+  end?: Date;
+  /**
+   * optional, used to determine how xAxis is formatted if `isGroupedByDate == true`
+   */
+  period?: string;
+  /**
+   * Formats dates as UTC?
+   */
+  utc?: boolean;
+  /**
+   * Inline styles
+   */
+  style?: React.CSSProperties;
+};
+
+class BaseChart extends React.Component<Props> {
+  static defaultProps = {
+    height: 200,
+    width: 'auto',
+    renderer: 'svg',
+    notMerge: true,
+    lazyUpdate: false,
+    onChartReady: () => {},
+    options: {},
+
+    series: [],
+    xAxis: {},
+    yAxis: {},
+    isGroupedByDate: false,
+  };
+
+  getEventsMap: ReactEchartProps['onEvents'] = {
+    click: (props, instance) => {
+      this.handleClick(props, instance);
+      this.props.onClick?.(props, instance);
+    },
+    highlight: (props, instance) => this.props.onHighlight?.(props, instance),
+    mouseover: (props, instance) => this.props.onMouseOver?.(props, instance),
+    datazoom: (props, instance) => this.props.onDataZoom?.(props, instance),
+    restore: (props, instance) => this.props.onRestore?.(props, instance),
+    finished: (props, instance) => this.props.onFinished?.(props, instance),
+  };
+
+  // TODO(ts): What is the series type? EChartOption.Series's data cannot have
+  // `onClick` since it's typically an array.
+  /**
+   * Handle series item clicks (e.g. Releases mark line or a single series item)
+   * This is different than when you hover over an "axis" line on a chart (e.g.
+   * if there are 2 series for an axis and you're not directly hovered over an item)
+   *
+   * Calls "onClick" inside of series data
+   */
+  handleClick = (series: any, instance: ECharts) => {
+    if (series.data) {
+      series.data.onClick?.(series, instance);
+    }
+  };
+
+  getColorPalette = () => {
```

Comment:
```suggestion
  getColorPalette() {
```
This doesn't need to be an arrow function its not an event handler.

Reply:
There's some reason I converted this, let me remember why 😬

Reply:
It was an arrow function before, but didn't need to be then either.